Puppet fixes and add Debian 11 support to the manual

ekohl opened the pull request:
Puppetserver (which our default installer scenario depends on) is only built for Debian 11. However, in our CI we only test with version 7 anyway so it's a good suggestion. It also raises the minimum version to 6 because that's what the installer requires. There is some Puppet 5 specific documentation in the Smart Proxy, but that code still works so I've left the documentation in.

ekohl committed:
Recommend Puppet 7

This also shortens the quickstart instructions. Rather than recommending
users, we're telling them. This is because no supported platform has
usable alternative packages. While there is one package in EPEL8 that
could work, it's built against Ruby 2.5 and can't be installed with Ruby
2.7.

ekohl committed:
Require Puppet 6

Since Foreman 3.0 the installer requires Puppet 6.15 or higher. This
adds a very late upgrade warning.

ekohl committed:
Avoid using apt-key in quickstart

In 19d2eb0d0c1b1451128d568f75d04a498d53a9fb the long installation
instructions were updated, but the quickstart was missed. This aligns
them.

Fixes: 19d2eb0d0c1b1451128d568f75d04a498d53a9fb

ekohl committed:
Add Debian 11 support to the manual

evgeni commented on _includes/manuals/nightly/2.1_quickstart_installation.md:
```diff
@@ -128,47 +129,67 @@ sudo yum -y install https://yum.theforeman.org/releases/{{page.version}}/el8/x86
+
+<div class="quickstart_os quickstart_os_debian11">
+  <p>
+    Using Puppet 7.x is recommended, which is available from Puppet's repository.
```
7 is the only supported one on Debian 11, should we rephrase that?

ekohl replied:
Something like this? I'd like to keep the text as equivalent as possible between the various distros. 
```suggestion
    Using Puppet 7.x from Puppet's repository is recommended.
```

Or, since it's the quickstart, say (the same as with the Foreman repo):
```suggestion
     Enable Puppet's 7.x repository:
```

And I do think we should write repository instead of repo on line 165.

evgeni replied:
Yeah 👍

ekohl replied:
Updated

ekohl committed:
Write repositories instead of repo

evgeni commented on _includes/manuals/nightly/2.1_quickstart_installation.md:
```diff
@@ -128,47 +127,61 @@ sudo yum -y install https://yum.theforeman.org/releases/{{page.version}}/el8/x86
+  <p>Enable the Foreman repositories:</p>
 
 {% highlight bash %}
+wget https://deb.theforeman.org/foreman.asc -O /etc/apt/trusted.gpg.d/foreman.asc
```
the later lines use `sudo`, so I guess this one should too?

evgeni commented on _includes/manuals/nightly/2.1_quickstart_installation.md:
```diff
@@ -128,47 +127,61 @@ sudo yum -y install https://yum.theforeman.org/releases/{{page.version}}/el8/x86
+{% highlight bash %}
+sudo apt-get -y install ca-certificates
+cd /tmp && wget https://apt.puppet.com/puppet7-release-bullseye.deb
+sudo dpkg -i /tmp/puppet7-release-bullseye.deb
```
```suggestion
sudo apt-get install /tmp/puppet7-release-bullseye.deb
```

evgeni commented on _includes/manuals/nightly/2.1_quickstart_installation.md:
```diff
@@ -128,47 +127,61 @@ sudo yum -y install https://yum.theforeman.org/releases/{{page.version}}/el8/x86
 
-    To use Puppet 6.x with Puppet Agent and Puppet server:
+{% highlight bash %}
+wget https://deb.theforeman.org/foreman.asc -O /etc/apt/trusted.gpg.d/foreman.asc
```
sudo

evgeni commented on _includes/manuals/nightly/2.1_quickstart_installation.md:
```diff
@@ -128,47 +127,61 @@ sudo yum -y install https://yum.theforeman.org/releases/{{page.version}}/el8/x86
-cd /tmp && wget https://apt.puppet.com/puppet6-release-focal.deb
-sudo dpkg -i /tmp/puppet6-release-focal.deb
+cd /tmp && wget https://apt.puppet.com/puppet7-release-focal.deb
+sudo dpkg -i /tmp/puppet7-release-focal.deb
```
```suggestion
sudo apt-get install /tmp/puppet7-release-focal.deb
```

evgeni commented on _includes/manuals/nightly/3.1.1_supported_platforms.md:
```diff
@@ -23,12 +23,14 @@ The following operating systems are supported by the installer, have packages an
   * Architectures: amd64
```
the line above this should read 20.04 Focal :)

ekohl committed:
Use apt-get instead of dpkg -i

Apt is able to install packages and dpkg should be considered an
internal tool.

ekohl committed:
Correct Ubuntu release name in supported platforms

Fixes: c7b949f60113e88a3157a2fb42e0a045aa19ae99

ekohl commented:
I think I've addressed them all now.
